src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/PostConfigureClientCredentialsOptions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Extensions.Options;
+
+namespace GSoft.Extensions.Http.Authentication.ClientCredentialsGrant;
+
+internal sealed class PostConfigureClientCredentialsOptions : IPostConfigureOptions<ClientCredentialsOptions>
+{
+    public void PostConfigure(string name, ClientCredentialsOptions options)
+    {
+        if (string.IsNullOrEmpty(options.CacheKey))
+        {
+            options.CacheKey = ComputeCacheKey(options);
+        }
+    }
+
+    private static string ComputeCacheKey(ClientCredentialsOptions options)
+    {
+        string hashedScopeHex;
+
+        using (var sha = SHA256.Create())
```

Comment:
In this scenario, the SHA256 will always be constant since the input bytes are the `Scope`. Is it the intended behaviour?

Also, if in this context you do not need a cryptographic secured has then it could be worth it to consider faster algorithms, especially if this gets called for every request (does not seem to be tho) the performance gain could be pretty significant. Since this seems to be used once at boot time, the `SHA256` computation time will not be an issue.

Reply:
Yes this is intentional. We want a unique cache key without any chance of collision with another unrelated JWT. That's why we also include the scope as part of the key.

But that cache key will literally be used in a database primary key such as SQL or Redis. So instead of using a user-defined, variable-length `Scope` (could be more than 100 characters), we use its hash representation to have a more consistent key length.

Performance is irrelevant because it only happens once during the app lifetime.

Reply:
Perfect, since it only happens once then performance is not an issue and predictability is what is wanted here. So no salt is required.